src/wallet/test/coinselector_tests.cpp

```diff
@@ -222,6 +224,17 @@ BOOST_AUTO_TEST_CASE(bnb_search_test)
     add_coin(1);
     vCoins.at(0).nInputBytes = 40; // Make sure that it has a negative effective value. The next check should assert if this somehow got through. Otherwise it will fail
     BOOST_CHECK(!testWallet.SelectCoinsMinConf( 1 * CENT, filter_standard, vCoins, setCoinsRet, nValueRet, coin_selection_params_bnb, bnb_used));
+
+    // Make sure that we aren't using BnB when there are preset inputs
+    empty_wallet();
+    add_coin(5 * CENT);
+    add_coin(3 * CENT);
+    add_coin(2 * CENT);
+    CCoinControl coin_control;
+    coin_control.fAllowOtherInputs = true;
+    coin_control.Select(COutPoint(vCoins.at(0).tx->GetHash(), vCoins.at(0).i));
+    BOOST_CHECK(testWallet.SelectCoins(vCoins, 10 * CENT, setCoinsRet, nValueRet, coin_control, coin_selection_params_bnb, bnb_used));
+    BOOST_CHECK(!bnb_used && !coin_selection_params_bnb.use_bnb);
```

Comment:
In commit "Test that BnB is not used when there are preset inputs"

Might want to split this up into two `BOOST_CHECK` statements, so if one condition fails, the output shows which one is failing.

Reply:
Done